src/main/java/org/opensearch/security/configuration/DlsFlsValveImpl.java

```diff
@@ -135,6 +137,10 @@ public DlsFlsValveImpl(
      */
     @Override
     public boolean invoke(PrivilegesEvaluationContext context, final ActionListener<?> listener) {
+        if (!isIndexPerm(context.getAction())) {
```

Comment:
I would guess that this needs to be at least also called for scroll requests, otherwise FLS and field masking might be not correctly applied. 

Generally, as the issue seems to be of a very specific kind, I am wondering whether the fix should be equally specific.

Reply:
I just pushed a commit to check the action prefix alone. If the action prefix is not `indices:`, then DlsFlsValveImpl would be skipped.

Reply:
Before #4380 , it would get short circuited [here](https://github.com/opensearch-project/security/blob/2.x/src/main/java/org/opensearch/security/configuration/DlsFlsValveImpl.java#L154-L156) because the `helpdesk_role` has no restrictions. The new code does not use that data structure and instead calls on `IndexNameExpressionResolver.getConcreteIndices` and passes in `*` because the resolved request shows `*` as you pointed out in https://github.com/opensearch-project/security/pull/4937#issuecomment-2499061401